RxSwiftWorkshop/Maps/Model/GeolocatorSpec.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Created by Krzysztof Siejkowski on 03/05/2017.
+// Copyright (c) 2017 Mobile Academy. All rights reserved.
+//
+
+import Foundation
+import Quick
+import Nimble
+import RxSwift
+import CoreLocation
+
+@testable
+import RxSwiftWorkshop
+
+class GeolocatorSpec: QuickSpec {
+
+    override func spec() {
+        describe("Geolocator") {
+
+            var disposeBag: DisposeBag?
+            var systemGeolocator: FakeSystemGeolocator?
+            var sut: Geolocator?
+
+            beforeEach {
+                disposeBag = DisposeBag()
+                let geolocator = FakeSystemGeolocator()
+                systemGeolocator = geolocator
+                sut = Geolocator(geolocator: geolocator)
+            }
+
+            afterEach {
+                disposeBag = nil
+                systemGeolocator = nil
+                sut = nil
+            }
+
+            it("should pass geolocation request to its SystemGeolocator") {
+                guard case let (disposeBag?, systemGeolocator?) = unwrap(disposeBag, systemGeolocator) else { return }
+                let address = "Valencia Airport"
+                sut?.geolocate(address: address).subscribe().disposed(by: disposeBag)
+                systemGeolocator.verifyCall(
+                    withIdentifier: FakeSystemGeolocator.Identifiers.geocodeAddressString, arguments: [address]
+                )
+            }
+
+            it("should return geolocation from its SystemGeolocator") {
```

Comment:
Frankly speaking I would do that differently.

First of all – break it down, this `it` seems to be too big. Secondly use RxTest, I think it might look better.

Something like (skipping UIOs, not relevant here): 

```swift
var scheduler: TestScheduler
var location: CLLocation?
var locationObserver: TestableObserver<CLLocation>

// somewhere above
scheduler = TestScheduler(initialClock: 0)

describe("SystemGeolocator return value") {

    beforeEach {
      guard case let (sut?, disposeBag?, systemGeolocator?) = unwrap(sut, disposeBag, systemGeolocator) else { return }
      // arrange
      location = CLLocation(latitude: 123.123, longitude: 132.132)
      systemGeolocator.geolocationObservable = scheduler.createColdObservable([next(0, location), completed(0)]).asObservable()
      // act
      locationObserver = scheduler.start {
          sut.geolocate(address: "Valencia Airport")
      }
    }

    it("should be valid geolocation") {
      // assert
      expect(locationObserver.events[0].value.element).to(equal(location))
    }
}
```

It applies to all below, as well.
It's not worth changing I guess, but good point for discussion.